### PR TITLE
Add Invalid Argument Throws to NSCharacterSet Initializers to match Reference Platform

### PR DESCRIPTION
**Pull request by MSFTFox**
NSCharacterSet characterSetWithBitmapRepresentation:nil and characterSetWithContentsOfFile:nil should throw illegal argument exceptions.

Two tests have been added to ensure this.

### CODE REVIEW

File: tests/unittests/Foundation/NSCharacterSetTests.m [hunk 1]
```diff
@@ -152,71 +152,89 @@ void testCharacter(NSCharacterSet* charSet, NSString* testString) {
...
+}
+
+ARM_DISABLED_TEST(NSCharacterSet, ExceptionTests) {
+    NSMutableCharacterSet* set;
+    @try {
+        set = [NSMutableCharacterSet characterSetWithBitmapRepresentation:nil];
+        EXPECT_TRUE_MSG(false, "NSCharacterSet should have thrown invalid argument exception.");
+    } @catch (NSException* e) {
+        ASSERT_OBJCEQ(e.name, @"NSInvalidArgumentException");
+    }
```
Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSCharacterSetTests.m]: although for arm expect any throw seems to work.

**Comment by msft-Jeyaram** [comment]
:shipit:

File: tests/unittests/Foundation/NSCharacterSetTests.m (hunk 1 above)
Comment (MSFTFox) [inline tests/unittests/Foundation/NSCharacterSetTests.m]: I'm not a fan of catching any throw. Sure it's expected that it's the correct throw but not guaranteed. It's more accurate to verify which exception was thrown.

Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSCharacterSetTests.m]: resonable

---
In reply to: [74654704](https://github.com/Microsoft/WinObjC/pull/877#discussion_r74654704) [](ancestors = 74654704)

**Comment by ms-jihua** [comment]
:shipit:

**Comment by aballway** [comment]
:shipit:

File: tests/unittests/Foundation/NSCharacterSetTests.m [hunk 2]
```diff
@@ -152,71 +152,89 @@ void testCharacter(NSCharacterSet* charSet, NSString* testString) {
...
+    @try {
+        set = [NSMutableCharacterSet characterSetWithBitmapRepresentation:nil];
+        FAIL();
+    } @catch (NSException* e) {
+        ASSERT_OBJCEQ(e.name, @"NSInvalidArgumentException");
+    }
+
+    @
+    try {
+        set = [NSMutableCharacterSet characterSetWithContentsOfFile:nil];
```
Comment (bbowman) [inline tests/unittests/Foundation/NSCharacterSetTests.m]: weird formatting. Double check that you don't have anything else weird preventing this from being correct. (clang-format uses a syntax tree form to evaluate how format which means weird formatting can sometimes be indicative of other errors).

**Comment by bbowman** [comment]
:shipit:

**Pull request merged**